components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -971,31 +970,27 @@ public AccessTokenDO getAccessToken(String accessTokenIdentifier, boolean includ
 
             if (includeExpired) {
                 if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled()) {
-                    if ((isTenantQualifiedUrlsEnabled && !isCrossTenantTokenInspectionAllowed)
-                            && tenantDomain != null) {
+                    if (!isCrossTenantTokenInspectionAllowed && tenantDomain != null) {
```

Comment:
do we get such instance whete tenant domain is null? can you put that ontop and throw error if the tenant domain is null?